### ISSUE TITLE
__slots__ not found on dataclass

### ISSUE THREAD

**Issue by AlexBlandin** [microsoft/pylance-release#4147]
## Environment data

Language Server version: Pylance language server 2023.3.30 (pyright 68a8e98a)
OS and version: Windows 10 21H2 19044.2728
Python version (& distribution if applicable, e.g. Anaconda): CPython 3.11.2 64-bit

## Code Snippet

```python
from dataclasses import dataclass

@dataclass(slots = True)
class Example:
  a: int
  b: int
  c: int
  
  def foo(self):
    return self.__slots__

print(Example(1, 2, 3).foo())
# ('a', 'b', 'c')

```

## Repro Steps

1. Enter code snippet
2. Observe red underline under `__slots__`
3. Mouse over and see it cannot be found despite being generated

## Expected behavior

`__slots__` gets detected at all. It providing useful type info based on the class would be a bonus, but not being a false error (shown with the `print`) is necessary.

## Actual behavior

```py
(function) __slots__: Unknown
Cannot access member "__slots__" for type "Example"
  Member "__slots__" is unknownPylancereportGeneralTypeIssues
```

## Logs

Nothing relevant.

**Comment by erictraut** [comment]
Yes, it's reasonable to expect that a type checker would synthesize the `__slots__` symbol for a dataclass that is constructed with the `slots=True` argument. This will be addressed in a future version of pylance.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023341-29-March-2023-PreRelease)
